third_party/nvfuser/csrc/scheduler/registry.cpp

```diff
@@ -1839,6 +1842,202 @@ class PersistentKernelScheduler : public SchedulerEntry {
     params_ = getPersistentHeuristics(fusion, runtime_info, data_cache);
     TORCH_INTERNAL_ASSERT(params_ != nullptr);
   }
+
+  static bool canScheduleRunTimeOuter(
+      Fusion* fusion,
+      SchedulerRuntimeInfo& runtime_info,
+      HeuristicSummary* data_cache,
+      const std::vector<TensorView*>& reduction_tvs,
+      const scheduler_utils::TvProperties& properties) {
+    FUSER_PERF_SCOPE("PersistentKernelScheduler::canScheduleRuntimeOuter");
+    FusionGuard fg(fusion);
+
+    const auto device_prop = at::cuda::getCurrentDeviceProperties();
+
+    const int64_t sm_register_file_size =
+        static_cast<int64_t>(device_prop->regsPerBlock * sizeof(int));
```

Comment:
Can't you preemptively subtract off the same values as in `getAvailableRegisterCount`?

Reply:
I think that should also work. Will consider in next iterations of PRs.